test/integration/tests/fuse.ts

```diff
@@ -0,0 +1,112 @@
+  let e2eCoord: TestEndtoEndCoordinator;
+  let doLogging: boolean;
+
+  before(async function () {
+    // Setup test environment and get contracts
+    const version = 1;
+    deployAddress = (await ethers.getSigners())[0].address;
+    if (!deployAddress) throw new Error(`No deploy address!`);
+
+    doLogging = Boolean(process.env.LOGGING);
+
+    const config = {
+      logging: doLogging,
+      deployAddress: deployAddress,
+      version: version
+    };
+
+    e2eCoord = new TestEndtoEndCoordinator(config, proposals);
+
+    doLogging && console.log(`Loading environment...`);
+    ({ contracts, contractAddresses } = await e2eCoord.loadEnvironment());
+    doLogging && console.log(`Environment loaded.`);
+  });
+
+  describe('D3 Fuse Plugin', function () {
+    it('deposit, claim, then withdraw', async function () {
+      const comptroller = await ethers.getContractAt('Unitroller', contractAddresses.rariPool8Comptroller);
+
+      const fD3 = await ethers.getContractAt(
+        'CErc20Delegator',
+        await comptroller.cTokensByUnderlying(contractAddresses.curveD3pool)
+      );
+
+      const d3Whale = '0x16c2bee6f55dab7f494dba643ff52ef2d47fba36';
+
+      await forceEth(d3Whale);
+
+      const d3 = await ethers.getContractAt('IERC20', contractAddresses.curveD3pool);
+
+      const signer = await getImpersonatedSigner(deployAddress);
+
+      doLogging && console.log('Seeding d3');
+      await d3
+        .connect(await getImpersonatedSigner(d3Whale))
+        .transfer(deployAddress, ethers.constants.WeiPerEther.mul(1_000_000));
+
+      doLogging && console.log('Approving d3');
+      await d3.connect(signer).approve(fD3.address, ethers.constants.MaxUint256);
+
+      const d3RewardsBalanceBefore = await contracts.convexD3poolRewards.balanceOf(plugin);
+      const fd3BalanceBefore = await fD3.balanceOfUnderlying(deployAddress);
+
+      doLogging && console.log('Minting fD3');
+      await fD3.connect(signer).mint(ethers.constants.WeiPerEther.mul(1_000_000));
+
+      expect(await contracts.convexD3poolRewards.balanceOf(plugin)).to.be.equal(
+        d3RewardsBalanceBefore.add(ethers.constants.WeiPerEther.mul(1_000_000))
+      );
+      expect(await fD3.balanceOfUnderlying(deployAddress)).to.be.equal(
+        fd3BalanceBefore.add(ethers.constants.WeiPerEther.mul(1_000_000))
+      );
+
+      doLogging && console.log('Redeeming Underlying');
+      await fD3.connect(signer).redeemUnderlying(ethers.constants.WeiPerEther.mul(1_000_000));
+
+      expect(await contracts.convexD3poolRewards.balanceOf(plugin)).to.be.equal(d3RewardsBalanceBefore);
+      expect(await fD3.balanceOfUnderlying(deployAddress)).to.be.equal(fd3BalanceBefore);
+
+      doLogging && console.log('Notifying rewards');
+
+      const operator = await getImpersonatedSigner(await contracts.convexD3poolRewards.operator());
+      await forceEth(await operator.getAddress());
+      await time.increaseTo(await contracts.convexD3poolRewards.periodFinish());
```

Comment:
it's not 100% sure that these e2e tests will hold over time, because sometimes the Convex rewards "stop flowing" and this will try to forward time in the past, but given the importance of d3pool now I guess we won't have this situation